eth/db/account.py

```diff
@@ -225,53 +221,23 @@ def state_root(self) -> Hash32:
 
     @state_root.setter
     def state_root(self, value: Hash32) -> None:
-        self._trie_cache.reset_cache()
-        self._trie.root_hash = value
+        if self._trie.root_hash != value:
+            self._trie_cache.reset_cache()
+            self._trie.root_hash = value
 
     def has_root(self, state_root: bytes) -> bool:
         return state_root in self._batchtrie
 
     #
     # Storage
     #
-    def get_storage(self, address: Address, slot: int, from_journal: bool=True) -> int:
-        validate_canonical_address(address, title="Storage Address")
-        validate_uint256(slot, title="Storage Slot")
-
-        account = self._get_account(address, from_journal)
-        storage = HashTrie(HexaryTrie(self._journaldb, account.storage_root))
-
-        slot_as_key = pad32(int_to_big_endian(slot))
```

Comment:
This line used to be succinct documentation of how int slots are turned into bytes keys. This PR scatters that knowledge across a couple methods in two different classes. What do you think of moving the `int_to_big_endian` part into `_decode_key` so that these can continue living side-by-side? It seems reasonable for `StorageLookup` to be the sole owner of how slots are read and written.

Reply:
Yeah, I feel it.

Unfortunately, the `BaseDB` API assumes `bytes` keys and values. We could *maybe* genericize it, but since the whole concept may get a rework in #1759 I'm hesitant to spend much time fiddling.

Luckily the int to bytes conversion is pretty straightforward. The "interesting" stuff like the pad32 and the keccak are still together.

Reply:
Alright, makes sense!